src/mozanalysis/metrics.py

```diff
@@ -413,6 +423,32 @@ def get_sanity_metrics(self, experiment_slug: str) -> list[Metric]:
                 ),
             ]
 
+        elif self.experiments_column_type == "events_stream":
+            return [
+                Metric(
+                    name=self.name + "_has_contradictory_branch",
+                    data_source=self,
+                    select_expr=agg_any(
+                        """IF(
+                JSON_VALUE(ds.event_extra, '$.experiment') = '{experiment_slug}',
+                JSON_VALUE(ds.event_extra, '$.branch'),
+                NULL
+            ) != e.branch """
```

Comment:
Yea I think you're right that this matches the other checks a little more closely than just checking the branch alone, and either way it's no harm to check the experiment + branch.